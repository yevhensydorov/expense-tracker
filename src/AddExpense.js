import React from 'react';

class AddExpense extends React.Component {
	constructor(props){
		super(props);
		this.state = {
			description: '',
			date: '',
			amount: ''
		};

		this.handleChange = this.handleChange.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
	}

	handleChange(event) {
		const value = event.target.value;
		const name = event.target.name; // find a name of the input
		this.setState({
			[name]: value // add the input name to the state obj
		});
	}

	handleSubmit(event) {
		event.preventDefault();
		// console.log(this.state);
		const expense = this.state;
		this.props.receiver(expense);
	}

	render() {
		return (
			<section className='add-block'>
				<form className='search-form' onSubmit={this.handleSubmit}>
					<div className='form-items'>

						<label 
							htmlFor="descr-input" 
							className='desrc-label'
						>
							Description
						</label>

						<input 
							onChange={this.handleChange}
							value={this.state.description}
							type='text'
							id='descr-input'
							className='descr-input'
							name='description'
						 />



						<label 
							htmlFor="date-input" 
							className='item-label'
						>
							Date
						</label>

						<input 
							onChange={this.handleChange}
							value={this.state.date}
							type='date'
							id='date-input'
							className='date-input'
							name='date'
						 />



						<label 
							htmlFor="amount-input" 
							className='item-label'
						>
							Amount
						</label>

						<input 
							onChange={this.handleChange}
							value={this.state.amount}
							type='text'
							id='amount-input'
							className='amount-input'
							name='amount'
						 />

					</div>

					<button className='submit-btn' type='submit'>Add</button>
					
				</form>
			</section>
		);
	}
}

export default AddExpense;