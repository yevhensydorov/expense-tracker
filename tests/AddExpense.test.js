import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import AddExpense from "../src/AddExpense";

describe("AddExpense", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<AddExpense />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("AddExpense", () => {
  it("captures change -DESCRIPTION- input and passes it event handler", () => {
    const event = {
      target: {
        value: "raspberry"
      }
    };
    const handleChange = jest.fn();
    const wrapper = shallow(<AddExpense handleChange={this.handleChange} />);
    // console.log(wrapper.find("input.descr-input").html());
    wrapper.find("input.descr-input").simulate("change", event);
    expect(handleChange.mock.calls).toEqual([["raspberry"]]);
  });
});

// describe("AddExpense", () => {
//   it("captures change -DATE- input and passes it event handler", () => {
//     const event = {
//       target: {
//         value: "2018-11-21"
//       }
//     };
//     const handleChange = jest.fn();
//     const wrapper = shallow(<AddExpense handleChange={handleChange} />);
//     wrapper.find("input.date-input").simulate("change", event);
//     expect(handleChange.mock.calls).toEqual([["2018-11-21"]]);
//   });
// });

// describe("AddExpense", () => {
//   it("captures change -DATE- input and passes it event handler", () => {
//     const event = {
//       target: {
//         value: "5"
//       }
//     };
//     const handleChange = jest.fn();
//     const wrapper = shallow(<AddExpense handleChange={handleChange} />);
//     wrapper.find("input.amount-input").simulate("change", event);
//     expect(handleChange.mock.calls).toEqual([["5"]]);
//   });
// });
