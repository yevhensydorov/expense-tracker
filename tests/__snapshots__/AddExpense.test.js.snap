// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddExpense matches the snapshot 1`] = `
<section
  className="add-block"
>
  <form
    className="search-form"
    onSubmit={[Function]}
  >
    <div
      className="form-items"
    >
      <label
        className="desrc-label"
        htmlFor="descr-input"
      >
        Description
      </label>
      <input
        className="descr-input"
        id="descr-input"
        name="description"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label
        className="item-label"
        htmlFor="date-input"
      >
        Date
      </label>
      <input
        className="date-input"
        id="date-input"
        name="date"
        onChange={[Function]}
        type="date"
        value=""
      />
      <label
        className="item-label"
        htmlFor="amount-input"
      >
        Amount
      </label>
      <input
        className="amount-input"
        id="amount-input"
        name="amount"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <button
      className="submit-btn"
      type="submit"
    >
      Add
    </button>
  </form>
</section>
`;
